### ISSUE TITLE
Dwarven Puzzle Room doesn't work correctly 

### ISSUE THREAD

**Issue by fizzet** [gemrb/gemrb#907]
#### Bug description
The puzzle in the Dwarven Puzzle Room (AR6010) seems to be broken. You're supposed to enter the three trigger tiles in order to open the exit, but GemRB opens it if you just move to the last of the three. The traps that should blast you with lightning bolts if you don't keep to the correct order don't fire, either.

#### Steps to reproduce
1. mta("ar6010")
2. move to the tile with the two intertwined rings at 10 o'clock in the innermost ring

#### Expected behavior
Exit should open only after correct procedure. Wrong steps should result in lightning blasts.

**Comment by lynxlynxlynx** [comment]
Hmm, can you do some bisecting to see if it ever worked?

**Comment by fizzet** [comment]
I still don't understand why the script works, but it does...

Now, the missing lightning strikes seem to be partly a problem of trajectory calculation (there's a FIXME in Projectile that might improve things:
https://github.com/gemrb/gemrb/blob/c4719a2d03e873de369d44a0988abea353799825/gemrb/core/Projectile.cpp#L890-L894
), and partly caused by https://github.com/gemrb/gemrb/blob/c4719a2d03e873de369d44a0988abea353799825/gemrb/core/Projectile.cpp#L1081
which makes party members exempt. I don't know enough to decide whether it's a problem with missing flags in the projectile itself, or a code issue.

**Comment by lynxlynxlynx** [comment]
Lightning is shot for me, do you mean the fact there's no payload? Comparing to bg2, lightb.pro is the same besides your bam change and higher speed. The spell uses the same projectile and appears to usually work (apart from bouncing #146 ). I don't think we differentiate any of this on engine version either.

I'll do a quick check where the payload is lost.

**Comment by fizzet** [comment]
It's not lost, it just misses its target. If I comment out the '^(GA_NO_ALLY|GA_NO_ENEMY)' it hits some of the time at least.

**Comment by lynxlynxlynx** [comment]
I see the problem, but it's clear it should also be hitting more often. We can probably get away by treating non-actor casters differently, since EA is not an applicable concept to them.

Even if you ctrl-j in many locations it just passes through. I can't check the original, but I doubt the projectile tracked the target there. So I'm pretty sure that lookup radius needs to be bumped, let's say to circle size / 4. Which would be 8-6px wide for most actors. Trying it out.

**Comment by lynxlynxlynx** [comment]
Pushed a fix, but not for the rare hits. With 10 it didn't work consistently enough, with 100 it did, but that's way too much (even 10 sounds high). Perhaps we have a geometry problem somewhere, but I'll leave you to explore further why some angles pretty consistently fail.

**Comment by bradallred** [comment]
the problem with not hitting enough of the time is explained by the `FIXME` @fizzet mentioned. We currently only check the radius around the bolt location. We should probably be doing a check at each node along the length of the projectile.

**Comment by lynxlynxlynx** [comment]
The iwd bolt is smaller, so I think the real problem is that the projectile is just too fast, so the steps are too big.

**Comment by bradallred** [comment]
I didn't realize the step size changed with speed, that seems odd.

**Comment by lynxlynxlynx** [comment]
In the instance I tested, not sure at what orientation, the step size was over 40px on both axes, but we sometimes moved twice as much, since there's also timing involved. @kmfrick wanted to port projectile movement to the new code paths, but we decided to postpone it to after the release.
